packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -248,73 +276,132 @@ const serviceMapFactory = (options: ServiceMapFactoryOptions) => {
+    });
+
+  const getBlockfrostTxSubmitProvider = () =>
+    new BlockfrostTxSubmitProvider({ blockfrost: getBlockfrostApi(), logger });
+
+  const getBlockfrostEnabledUtxoHttpService = () =>
+    new UtxoHttpService({
+      logger,
+      utxoProvider: new BlockfrostUtxoProvider({ blockfrost: getBlockfrostApi(), logger })
+    });
+
   return {
     [ServiceNames.Asset]: async () => {
-      const assetProvider = args.useTypeormAssetProvider ? getTypeormAssetProvider() : getDbSyncAssetProvider();
+      const chosenImplementation = args.useTypeormAssetProvider
+        ? ProviderImplementation.TYPEORM
+        : args.assetProvider ?? ProviderImplementation.DBSYNC;
+      const assetProvider = selectProviderImplementation<AssetProvider>(
+        chosenImplementation,
+        getTypeormAssetProvider,
+        getDbSyncAssetProvider,
+        getBlockfrostAssetProvider
+      );
       return new AssetHttpService({ assetProvider, logger });
     },
     [ServiceNames.StakePool]: async () => {
-      const stakePoolProvider = args.useTypeormStakePoolProvider
-        ? getTypeormStakePoolProvider()
-        : getDbSyncStakePoolProvider();
+      const chosenImplementation = args.useTypeormStakePoolProvider
+        ? ProviderImplementation.TYPEORM
+        : args.stakePoolProvider ?? ProviderImplementation.DBSYNC;
+
+      const stakePoolProvider = selectProviderImplementation<StakePoolProvider>(
+        chosenImplementation,
+        getTypeormStakePoolProvider,
+        getDbSyncStakePoolProvider
+      );
+
       return new StakePoolHttpService({ logger, stakePoolProvider });
     },
-    [ServiceNames.Utxo]: withDbSyncProvider(
-      async (dbPools, cardanoNode) =>
-        new UtxoHttpService({
-          logger,
-          utxoProvider: new DbSyncUtxoProvider({
-            cache: {
-              healthCheck: healthCheckCache
-            },
-            cardanoNode,
-            dbPools,
-            logger
-          })
-        }),
-      ServiceNames.Utxo
-    ),
-    [ServiceNames.ChainHistory]: withDbSyncProvider(async (dbPools, cardanoNode) => {
-      const metadataService = createDbSyncMetadataService(dbPools.main, logger);
-      const chainHistoryProvider = new DbSyncChainHistoryProvider(
-        { paginationPageSizeLimit: args.paginationPageSizeLimit! },
-        {
-          cache: {
-            healthCheck: healthCheckCache
-          },
-          cardanoNode,
-          dbPools,
-          logger,
-          metadataService
-        }
-      );
-      return new ChainHistoryHttpService({ chainHistoryProvider, logger });
-    }, ServiceNames.ChainHistory),
+    [ServiceNames.Utxo]:
+      args.utxoProvider === ProviderImplementation.BLOCKFROST
+        ? getBlockfrostEnabledUtxoHttpService
+        : withDbSyncProvider(
+            async (dbPools, cardanoNode) =>
+              new UtxoHttpService({
+                logger,
+                utxoProvider: new DbSyncUtxoProvider({
```

Comment:
All provider implementations use the same HttpService. I think it would be cleaner if we switch only the provider
```suggestion
                utxoProvider: selectProviderImplementation(args.utxoProvider, ...)
```

Reply:
Done here 0d44798bfa387540e2db07bfab967eab043029dd